### PR TITLE
Fetch TAU samples from TAU page

### PR DESCRIPTION
**Pull request by grzegorz-czajkowski**
[Issue] http://suprem.sec.samsung.net/jira/browse/TIZENWF-2567
[Problem] WATT does not have all TAU samples
[Solution] Fetch TAU samples from https://samsung.github.io/TAU/examples/
           while requesting /demos API with sample path as query parameter, for example
           http://localhost:3000/demos/?path=mobile%2FUIComponents%2Fcomponents%2Fcontrols%2Fcheckbox.html
           Since wget is used to download page, curl has been removed in favor of wget.
[TEST]
    1. Create sample html with the following content:
      <html>
        <body>
          <script>
            window.open("http://localhost:3000/demos/?path=" + encodeURIComponent("mobile/UIComponents/components/controls/checkbox.html"));
          </script>
        </body>
      </html>
      Alternatively, you can manually encode path by replacing '/' with '%2F'.

    2. Brackets project should be opened without user authentication.

[Remarks]
    white page while downloading sample (this takes arround ~20 second)
    DE does not start - lack of config.xml
    missing images in tau, wget error checking is skiped now

Signed-off-by: Grzegorz Czajkowski <g.czajkowski@samsung.com>

### CODE REVIEW

File: launch [hunk 1]
```diff
@@ -70,18 +70,12 @@ def import_additional():
 import_additional()
 
 def install_nodejs():
-  if os_name == 'Darwin':
```
Comment (lslachciak) [inline launch]: why remove MAC related lines?
on MAC there is no wget https://www.mkyong.com/mac/wget-on-mac-os-x/

File: routes/demos.js [hunk 2]
```diff
@@ -42,15 +45,23 @@ module.exports = function (express) {
...
-              return callback(err);
+          const tauExamplesHost = config.get('TAUExamplesHost');
+          const samplePath = req.query.path;
+          if (!tauExamplesHost || !samplePath)
+            return callback(`Could not resolve path: ${tauExamplesHost}${samplePath}`);
+
+          const sampleUrl = new URL(samplePath, tauExamplesHost);
+          // Define number of unnecessary directories to be omitted while downloading.
+          const numDirsToCut = sampleUrl.pathname.match(/TAU\/examples\/mobile|werable\/UIComponents/g) ? 4 : 0;
+          exec(`wget --page-requisites --convert-links --no-host-directories --cut-dirs=${numDirsToCut} --directory-prefix ${projectPath} ${sampleUrl}`, (error, stdout, stderr) => {
```
Comment (lslachciak) [inline routes/demos.js]: I suggest adding TODO comment to consider using 
https://www.npmjs.com/package/node-wget
or
https://www.npmjs.com/package/wgetjs

File: routes/demos.js [hunk 3]
```diff
@@ -42,15 +45,23 @@ module.exports = function (express) {
...
+            if (error) {
+              // FIXME: background-image's url from tau.css references file that
+              // does not exist on server. Uncomment line below once it's fixed.
+              // return callback(error);
```
Comment (lslachciak) [inline routes/demos.js]: let's put console.log here so we don't miss it in future

File: routes/demos.js (hunk 2 above)
Comment (grzegorz-czajkowski) [reply routes/demos.js]: Both packages do not support options that we are using:
"--page-requisites --convert-links --no-host-directories --cut-dirs=${numDirsToCut}"

File: launch (hunk 1 above)
Comment (grzegorz-czajkowski) [reply launch]: Restored

File: routes/demos.js (hunk 3 above)
Comment (grzegorz-czajkowski) [reply routes/demos.js]: Done.

File: routes/demos.js (hunk 2 above)
Comment (grzegorz-czajkowski) [reply routes/demos.js]: I found promising candidate https://github.com/website-scraper/node-website-scraper

**Pull request merged**